jnim/private/jni_export.nim

```diff
@@ -429,9 +437,9 @@ macro jexport*(a: varargs[untyped]): untyped =
       assert(false)
 
   block: # Finalizer thunk
-    let thunkName = genSym(nskProc, JniExportedFunctionPrefix & className & "__0")
+    let thunkName = ident(JniExportedFunctionPrefix & className & "_" & FinalizerName.replace("_", "_1"))
```

Comment:
I don't understand what this `replace` does.

Reply:
https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/design.html#wp133 "We adopted a simple name-mangling scheme to ensure that all Unicode characters translate into valid C function names. (...)"